### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by HanaSabih**
Start building your school library app by implementing the classes to represent students and teachers.
 

- [x] Add Rubocop linter for ruby
- [x]  Add Person class from person.rb
- [x]  Add Student class from student.rb
- [x]  Add Teacher class from teacher.rb
- [x]  Update README.md file to reflect project

### CODE REVIEW

File: ruby/person.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Person
+  attr_accessor :name, :age, :parent_permission
+  attr_reader :id
+
+  def initialize(name, age = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
```
Comment (bobb-Rob) [inline ruby/person.rb]: - [ ] Based don the project requirement, the constructor parameters that should be optional are ` name` and `parent_permission` but you are making `age` optional. Kindly fix this.


image

File: ruby/person.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    of_age? || parent_permission
```
Comment (bobb-Rob) [inline ruby/person.rb]: - [ ] On line 17, the `parent_permission` instance variable should be accessed using the `@` sign like so 👇🏾 

```ruby 
of_age? || @parent_permission
```

File: ruby/person.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  def initialize(name, age = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def of_age?
+    @age >= 18
+  end
```
Comment (bobb-Rob) [inline ruby/person.rb:16]: - [ ] This method `of_age?` should be a private method kike so 👇🏾 

```ruby
  def can_use_services? 
    of_age?() || @parent_permission
  end


private

  def of_age?
    @age >= 18
  end
```

File: ruby/student.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+require_relative 'Person'
+class Student < Person
+  attr_accessor :classroom
+
+  def initialize(classroom, name, age = 'Unknown', parent_permission: true)
```
Comment (bobb-Rob) [inline ruby/student.rb]: - [ ] here also, `:name` is optional and not `:age` . Please fix this

File: ruby/teacher.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+require_relative 'Person'
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(specialization, name, age = 'Unknown', parent_permission: true)
```
Comment (bobb-Rob) [inline ruby/teacher.rb]: - [ ] Same here, `:name` should be the parameter with a default value and not the `:age`. Kindly fix this.

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi @HanaSabih , 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)
✅ Neat Commit messages

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by addod19 — APPROVED** [review]
### Hi  @HanaSabih, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [ ] Great job fixing all the required changes :rocket: 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**